### DISCUSSION TITLE
How to get the Parameter Annotations?

### DISCUSSION THREAD

**Discussion by anjunar** [scalacenter/tasty-query#344] [Ideas]
Is the a way to get the Parameter Annotations of a Method?

I would say that a ParameterSymbol is missing.

**Comment by sjrd** [comment]
You should be able to go through the `tree` of the method symbol:
```scala
val methodSym: TermSymbol = ???
methodSym.tree match
  case Some(DefDef(_, paramLists, _, _)) =>
    val paramSymss = paramLists.map {
      case Left(valDefs) => valDefs.map(_.symbol)
      case Right(typeParams) => typeParams.map(_.symbol)
    }
  case _ =>
    println("hum, I don't have a tree, or it's not a method
```
That only works for methods coming from Scala 3, though.

**Reply by anjunar** [reply]
Very well... Thank you :-)

**Reply by anjunar** [reply]
What would I do when analysing Java classes?

I guess the tree is then not available.

**Reply by bishabosha** [reply]
It will be in the classfile if the retention is set that way, but our handling of annotations from Java is pretty minimal so far

**Reply by anjunar** [reply]
Hmmm... I have a mixed Project of Java and Scala, because I am using Hibernate.

So will there be a version that supports Java Annotations as well?

**Reply by sjrd** [reply]
I guess eventually we should support them.
